ui/components/component-library/button-base/button-base.js

```diff
@@ -68,19 +70,27 @@ export const ButtonBase = ({
       {...props}
     >
       <Text
-        as="span"
         className="mm-button-base__content"
+        as="div"
         justifyContent={JustifyContent.center}
         alignItems={AlignItems.center}
         gap={2}
         variant={TextVariant.bodyMd}
         color={TextColor.inherit}
+        ellipsis={ellipsis}
         {...textProps}
       >
         {startIconName && (
           <Icon name={startIconName} size={Size.SM} {...startIconProps} />
         )}
-        {children}
+        <Text
+          as="div"
```

Comment:
Not sure this is semantically correct? Maybe it would be better to make this a one off in the component @darkwing is working on? Couldn't he just do these updates using the `textProps` object?
```
textProps={{ ellipsis: true, as: 'div' }}
```
https://stackoverflow.com/questions/12982269/is-it-semantically-incorrect-to-put-a-div-or-span-inside-of-a-button

<img width="763" alt="Screenshot 2023-03-16 at 4 39 40 PM" src="https://user-images.githubusercontent.com/8112138/225775839-62437dce-1584-4290-84d9-bc9b4b0d14ae.png">